vars/copyProjectConfig.groovy

```diff
@@ -4,15 +4,13 @@ def call(projectPath, lvVersion){
    echo "Copying configuration file for $projectPath"
    configFileName = "$projectPath" + ".config"
    
-   def currentVersion = lvVersion as int
-   def newAssemblyVersion = "${currentVersion}.0.0.0"
-   
-   def previousVersion = currentVersion - 1
-   def oldAssemblyVersion = "0.0.0.0-${previousVersion}.9.9.9"
-   
+   def defaultVersion = ["$lvVersion": "${lvVersion}.0.0.0"]
+   def assemblyVersions = readProperties defaults: defaultVersion, file: "niveristand-custom-device-build-tools/resources/assemblyVersions.properties"
+
+   def newAssemblyVersion = assemblyVersions."$lvVersion"
+
    def fileContent = readFile "niveristand-custom-device-build-tools/resources/LabVIEW.exe.config"
-   fileContent = fileContent.replaceAll("(oldVersion=\")[^\"]+","\$1$oldAssemblyVersion")
-      .replaceAll("(newVersion=\")[^\"]+","\$1$newAssemblyVersion")
-   
+   fileContent = fileContent.replaceAll("(newVersion=\")[^\"]+","\$1$newAssemblyVersion")
```

Comment:
Can you add a link to a saved regex101 query that shows how this regex works?

Reply:
https://regex101.com/r/tBNE56/1
The replacement just uses the first matched group followed by the new assembly version and a `"` char.